File: frontend/types/document.ts
export interface DocumentResponse {
  id: number;
  title: string;
  original_filename: string;
  file_path: string;
  status: string;
  uploaded_at: string;
}

export interface OCRResultResponse {
  document_id: number;
  status: string;

  ocr_text?: string;
  translated_text?: string;
  summary?: string;
  error_message?: string;
}

export interface ChatRequest {
  question: string;
}

export interface ChatResponse {
  document_id: number;
  question: string;
  answer: string;
}

File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: int
    title: str
    original_filename: str
    file_path: str
    status: str
    uploaded_at: datetime

    class Config:
        from_attributes = True


class OCRResultResponse(BaseModel):
    document_id: int
    status: str

    ocr_text: str | None = None

    translated_text: str | None = None

    summary: str | None = None

    error_message: str | None = None